clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/client/BlockingStatusChecker.groovy

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright 2016 Target, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.clouddriver.openstack.client
+
+import com.netflix.spinnaker.clouddriver.openstack.deploy.exception.OpenstackProviderException
+import groovy.transform.PackageScope
+
+import java.util.concurrent.TimeUnit
+
+/**
+ * This class abstracts the algorithm to continually poll until a status is obtained or a timeout occurs.  Openstack
+ * API requires that the load balancer be in an ACTIVE state for it to create associated relationships (i.e. listeners, pools,
+ * monitors).  Each modification will cause the load balancer to go into a PENDING state and back to ACTIVE once the change
+ * has been made.  Depending on your implementation, the timeout and polling intervals would need to be tweaked.
```

Comment:
This is only used by load balancer now, but to your point, could be used in other areas.   I have decided to remove the from method with defaults and make it configuration based.  Therefore, the client using the checker will be responsible for providing timeout and interval values.  

Configuration will contain a reasonable default for stable installations,  Right now we are using devstack which is undersized and takes approximately 7 or 8 minutes to launch a load balancer.  These defaults could change as we get a stable liberty install.